### PR TITLE
[CHK-1734] feat(api): add endpoint for creating form for a payment method

### PR DESCRIPTION
**Pull request by giovanniberti**
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-spec/api.yaml [hunk 1]
```diff
@@ -146,6 +146,45 @@ paths:
               schema:
                 $ref: '#/components/schemas/ProblemJson'
 
+  /payment-methods/{id}/dubidubidu:
```
Comment (giovanniberti) [inline api-spec/api.yaml:149]: 🧵 for choice of endpoint name

File: api-spec/api.yaml [hunk 2]
```diff
@@ -434,6 +473,51 @@ components:
...
+          items:
+            $ref: '#/components/schemas/Field'
+      required:
+        - paymentMethod
+        - form
+    Field:
+      type: object
+      properties:
+        type:
+          type: string
```
Comment (pietro-tota) [inline api-spec/api.yaml:510]: What do you think about add format: uri specification on those fields?

**Review by pietro-tota — DISMISSED** [review]

File: api-spec/api.yaml (hunk 2 above)
Comment (giovanniberti) [reply api-spec/api.yaml:510]: Fixed in ee6c47a

**Review by pietro-tota — APPROVED** [review]

File: src/main/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsController.java [hunk 3]
```diff
@@ -183,4 +183,12 @@ public Mono<ResponseEntity<CalculateFeeResponseDto>> calculateFees(
                 ResponseEntity::ok
         );
     }
+
+    @Override
+    public Mono<ResponseEntity<PreauthorizationResponseDto>> createPreauthorization(
+                                                                                    String id,
+                                                                                    ServerWebExchange exchange
+    ) {
+        return Mono.just(ResponseEntity.status(HttpStatus.METHOD_NOT_ALLOWED).build());
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsController.java:192]: 🚲 maybe we can use a not implemented yet error code here?

Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsController.java:192]: [Looked into that](https://stackoverflow.com/questions/32889983/is-http-501-appropriate-for-an-unimplemented-api) 🙂
`501 Not Implemented` should be used only when a method is not recognized by the server indipendently from the resource, while `405 Method Not Allowed` is the correct HTTP status code if a request cannot be made to a resource with a method (technically the endpoint should always return an `Allow` header with a list of allowed methods when using `405`, but given that this implementation won't ever go to production I think it's safe to not be 100% compliant)

**Pull request merged**